examples/src/main/java/com/mapbox/navigation/examples/core/WaypointsRerouteActivity.kt

```diff
@@ -0,0 +1,361 @@
+import com.mapbox.navigation.ui.camera.NavigationCamera
+import com.mapbox.navigation.ui.map.NavigationMapboxMap
+import java.lang.ref.WeakReference
+//import kotlinx.android.synthetic.main.activity_reroute_layout.*
+import kotlinx.android.synthetic.main.activity_reroute_layout.container
+import kotlinx.android.synthetic.main.activity_reroute_layout.mapView
+import kotlinx.android.synthetic.main.activity_reroute_layout.startNavigation
+//import kotlinx.android.synthetic.main.activity_waypoints_reroute_layout.*
+import timber.log.Timber
+
+/**
+ * This activity shows how to:
+ * - observe reroute events with the Navigation SDK's [RoutesObserver];
+ * - replace default [RerouteController] and handle re-route events.
+ */
+class WaypointsRerouteActivity : AppCompatActivity(), OnMapReadyCallback, OffRouteObserver,
+        RerouteController.RerouteStateObserver {
+
+    companion object {
+        private const val DEFAULT_FASTEST_INTERVAL = 500L
+        private const val DEFAULT_ENGINE_REQUEST_INTERVAL = 1000L
+    }
+
+    private var mapboxNavigation: MapboxNavigation? = null
+    private var navigationMapboxMap: NavigationMapboxMap? = null
+    private var directionRoute: DirectionsRoute? = null
+    private val routeSettings = CoordinatesHolder()
+    private val stopsController = StopsControllerr()
+
+    private val locationListenerCallback = MyLocationEngineCallback(this)
+    private val tripSessionStateObserver = object : TripSessionStateObserver {
+        override fun onSessionStateChanged(tripSessionState: TripSessionState) {
+            when (tripSessionState) {
+                TripSessionState.STARTED -> {
+                    startNavigation.visibility = GONE
+                    stopLocationUpdates()
+                    mapboxNavigation?.registerRouteProgressObserver(routeProgressObserver)
+                }
+                TripSessionState.STOPPED -> {
+                    stopsController.clear()
+                    startLocationUpdates()
+                    navigationMapboxMap?.removeRoute()
+                    updateCameraOnNavigationStateChange(false)
+                }
+            }
+        }
+    }
+
+    private val routeObserver = object : RoutesObserver {
+        override fun onRoutesChanged(routes: List<DirectionsRoute>) {
+            startNavigation.visibility = if (routes.isNotEmpty()) {
+                VISIBLE
+            } else {
+                GONE
+            }
+            directionRoute = routes.firstOrNull()
+            navigationMapboxMap?.drawRoutes(routes)
+        }
+    }
+    private val routeProgressObserver = object : RouteProgressObserver {
+        override fun onRouteProgressChanged(routeProgress: RouteProgress) {
+        }
+    }
+    private val routesReqCallback = object : RoutesRequestCallback {
+        override fun onRoutesReady(routes: List<DirectionsRoute>) {
+            Timber.d("route request success %s", routes.toString())
```

Comment:
```suggestion
            Timber.d("route request success   ${routes.toString()}")
```
kotlin 😎 here and below